### PR TITLE
added documentation, js standard style and jest testing to kadane algorithm file.

### PR DESCRIPTION
**Pull request by thisabhijeet**
Added documentation, js standard style and jest testing to kadane algorithm file.

# Welcome to JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new JavaScript files are placed inside an existing directory.
* [x] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Dynamic-Programming/KadaneAlgo.js [hunk 1]
```diff
@@ -1,20 +1,35 @@
...
+    if (maxSum < cummulativeSum) {
       maxSum = cummulativeSum
+    } else if (cummulativeSum < 0) {
+      cummulativeSum = 0
     }
   }
   return maxSum
   // This function returns largest sum contiguous sum in a array
 }
 function main () {
```
Comment (raklaptudirm) [inline Dynamic-Programming/KadaneAlgo.js:26]: Remove the main function.

File: Dynamic-Programming/KadaneAlgo.js [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
+  // input array
   const myArray = [1, 2, 3, 4, -6]
-  const result = KadaneAlgo(myArray)
+  // calling the function
+  const result = kadaneAlgo(myArray)
+  // result is the variable for storing the ourput of kadaneAlgo function
   console.log(result)
 }
 main()
+module.exports = { kadaneAlgo }
```
Comment (raklaptudirm) [inline Dynamic-Programming/KadaneAlgo.js:35]:
```suggestion
export { kadaneAlgo }
```
Use ESM

File: Dynamic-Programming/tests/KadaneAlgo.test.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const fc = require('../kadaneAlgo')
```
Comment (raklaptudirm) [inline Dynamic-Programming/tests/KadaneAlgo.test.js:1]:
```suggestion
 import { kadaneAlgo } '../KadaneAlgo'
```
Use ESM and the correct file name.

File: Dynamic-Programming/tests/KadaneAlgo.test.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+test('test1', () => {
```
Comment (raklaptudirm) [inline Dynamic-Programming/tests/KadaneAlgo.test.js:2]: Add a better description of the test that `test1`.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

**Comment by thisabhijeet** [comment]
import { kadaneAlgo } from '../KadaneAlgo' :- i have used this in the test file. Not using 'keyword' was creating an error

**Comment by raklaptudirm** [comment]
Please fix your tests

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**